src/sql.rs

```diff
@@ -1288,6 +1288,57 @@ async fn open(
             update_icons = true;
             sql.set_raw_config_int(context, "dbversion", 66).await?;
         }
+        if dbversion < 67 {
+            info!(context, "[migration] v67");
+            for prefix in &["", "configured_"] {
+                if let Some(server_flags) = sql
+                    .get_raw_config_int(context, format!("{}server_flags", prefix))
+                    .await
+                {
+                    let imap_socket_flags = server_flags & 0x700;
+                    let imap_security_key = format!("{}mail_security", prefix);
+                    match imap_socket_flags {
+                        0x100 => {
+                            sql.set_raw_config_int(context, imap_security_key, 2)
+                                .await?
+                        } // STARTTLS
+                        0x200 => {
+                            sql.set_raw_config_int(context, imap_security_key, 1)
+                                .await?
+                        } // SSL/TLS
+                        0x400 => {
+                            sql.set_raw_config_int(context, imap_security_key, 3)
+                                .await?
+                        } // Plain
+                        _ => {
+                            sql.set_raw_config_int(context, imap_security_key, 0)
+                                .await?
+                        }
+                    }
```

Comment:
Something seems to be wrong here to me (or I am missing sth) - e.g. [Plain is 3, not 0 (??)](https://github.com/deltachat/deltachat-core-rust/blob/bebd9497c7d607c8221baa84fad16837229684b9/src/provider/mod.rs#L29). Also, `Socket::Plain as i32` looks nicer than a magic number IMHO.

Same below for smtp.

Reply:
0x400 is converted to 3. The comment goes after the 0x400 case because rustfmt placed it there.

Reply:
i was also stumbling upon these lines, i think, this is a one of the few examples where rustfmt makes things less readable and hard to follow.

a pragmatic fix is to use a shorter name for the key - then things go to one line again, example: https://github.com/deltachat/deltachat-core-rust/pull/1841